packages/contentstack-bulk-publish/src/producer/publish-entries.js

```diff
@@ -92,13 +113,32 @@ async function getEntries(stack, contentType, locale, bulkPublish, environments,
           bulkPublishSet = [];
           return resolve();
         }
-        await getEntries(stack, contentType, locale, bulkPublish, environments, apiVersion, skipCount);
+        await getEntries(stack, contentType, locale, bulkPublish, environments, apiVersion, variantsFlag, skipCount);
         return resolve();
       })
       .catch((error) => reject(error));
   });
 }
 
+async function getVariantEntries(stack, contentType, entries, index, queryParams) {
+  try {
+    const variantsEntriesResponse = await stack
+      .contentType(contentType)
+      .entry(entries[index].uid)
+      .variants()
+      .query(queryParams)
+      .find();
+
+    const variants = variantsEntriesResponse.items.map(entry => ({
+      uid: entry.variants_uid,
+    }));
+    
+    return variants;
+  } catch (error) {
+    throw new Error(`Error fetching variants: ${error.message}`);
```

Comment:
@sunil-lakshman , From api side we get error message in errorMessage, message and errors properties so it's better to handle like this
error?.errorMessage || error?.message || error. Pls check other plugin as well